modules/python/test/test_misc.py

```diff
@@ -50,39 +68,575 @@ class Arguments(NewOpenCVTests):
+                        type(e).__name__, e, a[i],
+                        type(a[i]).__name__
+                    )
+                )
+            else:
+                self.assertEqual(expected, actual,
+                                 msg="Can't convert array element {0} of type {1}\n"
+                                 "Expected: {2} vs Actual: {3}".format(
+                                     a[i],
+                                     type(a[i]).__name__,
+                                     expected, actual
+                                 ))
+
+        with self.assertRaises(
+            TypeError,
+            msg='Exception is not risen for tuple slice with 1 element'
+        ):
+            _ = cv.utils.dumpBool((True, False, False)[0:1])
+
+        for not_convertible in (1.2, np.float(2.3), 's', 'str', (1, 2),
+                                [1, 2], a, 1 + 1j, 2j, np.complex(1.1),
+                                np.complex(-0.5, 1.), np.complex(imag=3j)):
+            with self.assertRaises(
+                (TypeError, OverflowError),
+                msg='Exception is not risen for {} of type {}'.format(
+                    not_convertible, type(not_convertible).__name__
+                )
+            ):
+                _ = cv.utils.dumpBool(not_convertible)
+
+    @unittest.skip('Wrong conversion behavior')
+    def test_parse_to_bool_extra(self):
+        _, max_size_t = get_limits(ctypes.c_size_t)
+        for convertible in (-1, max_size_t):
+            expected = 'Bool: {}'.format(bool(convertible)).lower()
+            try:
+                actual = cv.utils.dumpBool(convertible).lower()
+            except Exception as e:
+                self.fail(
+                    '{} "{}" is risen for conversion {} of type {}'.format(
+                        type(e).__name__, e, convertible,
+                        type(convertible).__name__
+                    )
+                )
+            else:
+                self.assertEqual(expected, actual,
+                                 msg="Can't convert {0} of type {1}\n"
+                                 "Expected: {2} vs Actual: {3}".format(
+                                     convertible,
+                                     type(convertible).__name__,
+                                     expected, actual
+                                 ))
+
+        for not_convertible in (np.array([False]),
+                                np.array([True], dtype=np.bool)):
+            with self.assertRaises(
+                (TypeError, OverflowError),
+                msg='Exception is not risen for {} of type {}'.format(
+                    not_convertible, type(not_convertible).__name__
+                )
+            ):
+                _ = cv.utils.dumpBool(not_convertible)
+
+    def test_parse_to_int(self):
+        min_int, max_int = get_limits(ctypes.c_int)
+        for convertible in (-10, -1, 2, True, False, int(43.2), np.uint8(15),
```

Comment:
> True, False

Not sure that we really want them in this list (at least for type-safety and better overloads handling - we have bool+int parameters in a row).

/cc @mshabunin

Reply:
I added boolean to convertible list to state implicit conversion behavior.

I think we should leave overload resolution to C++ backend. For example, converter will indicate if such conversion occurred.